test/unit/bundle.test.js

```diff
@@ -2712,10 +2713,131 @@ describe('bundle files method - 3.0', function () {
+  });
+
+  it('Should return bundled file as json with deep url refs - remote_url_refs', async function () {
+    let contentRootFile = fs.readFileSync(remoteURLRefExamples + '/root.json', 'utf8'),
+      spacecraft = fs.readFileSync(remoteURLRefExamples + '/schemas/Spacecraft.json', 'utf8'),
+      peakThrustSecond = fs.readFileSync(remoteURLRefExamples + '/schemas/peakThrustSecond.json', 'utf8'),
+      peakThrustSecondProperty =
+        fs.readFileSync(remoteURLRefExamples + '/schemas/peakThrustSecondProperty.json', 'utf8'),
+      remoteRefResolver = async (refURL) => {
+        if (refURL.includes('peakThrustSecondProperty')) {
+          return JSON.parse(peakThrustSecondProperty);
+        }
+
+        if (refURL.includes('peakThrustSecond')) {
+          return JSON.parse(peakThrustSecond);
+        }
+
+        if (refURL.includes('Spacecraft')) {
+          return JSON.parse(spacecraft);
+        }
+      },
+      expected = fs.readFileSync(remoteURLRefExamples + '/expected.json', 'utf8'),
+      input = {
+        type: 'multiFile',
+        specificationVersion: '3.0',
+        rootFiles: [
+          {
+            path: 'root.json'
+          }
+        ],
+        data: [
+          {
+            path: 'root.json',
+            content: contentRootFile
+          }
+        ],
+        options: {},
+        bundleFormat: 'JSON',
+        remoteRefResolver
+      };
+
+    const res = await Converter.bundle(input);
+
+    expect(res).to.not.be.empty;
+    expect(res.result).to.be.true;
+    expect(res.output.specification.version).to.equal('3.0');
+    expect(JSON.stringify(JSON.parse(res.output.data[0].bundledContent), null, 2)).to.be.equal(expected);
+  });
+
+  it('Should return bundled file as json with yaml content remote refs - remote_url_refs', async function () {
+    let contentRootFile = fs.readFileSync(remoteURLRefExamples + '/yaml/root.json', 'utf8'),
+      spacecraftId = fs.readFileSync(remoteURLRefExamples + '/yaml/SpacecraftId.yaml', 'utf8'),
+
+      remoteRefResolver = async (refURL) => {
+        if (refURL.includes('SpacecraftId')) {
+          return spacecraftId;
+        }
+      },
+      expected = fs.readFileSync(remoteURLRefExamples + '/yaml/expected.json', 'utf8'),
+      input = {
+        type: 'multiFile',
+        specificationVersion: '3.0',
+        rootFiles: [
+          {
+            path: 'root.json'
+          }
+        ],
+        data: [
+          {
+            path: 'root.json',
+            content: contentRootFile
+          }
+        ],
+        options: {},
+        bundleFormat: 'JSON',
+        remoteRefResolver
+      };
+
+    const res = await Converter.bundle(input);
+
+    expect(res).to.not.be.empty;
+    expect(res.result).to.be.true;
+    expect(res.output.specification.version).to.equal('3.0');
+    expect(JSON.stringify(JSON.parse(res.output.data[0].bundledContent), null, 2)).to.be.equal(expected);
+  });
```

Comment:
Let's add few more tests to make sure the below scenarios also works.

- Circular refs (A -> B -> C -> A). We should have this handled and circular ref should not be resolved to avoid obvious further issues.
- If remoteRefResolver throws error, depending upon our handling expect the result instead entire function throwing error.
- Should return bundled file as yaml with yaml remote refs
- Should return bundled file as yaml with json content remote refs - remote_url_refs